rosidl_typesupport_opensplice_cpp/include/rosidl_typesupport_opensplice_cpp/impl/error_checking.hpp

```diff
@@ -0,0 +1,271 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef __rosidl_typesupport_openslice_cpp__impl__error_checking__h__
+#define __rosidl_typesupport_openslice_cpp__impl__error_checking__h__
+
+#include <ccpp_dds_dcps.h>
+
+#include "builtin_interfaces/msg/duration__struct.hpp"
+
+namespace rosidl_typesupport_opensplice_cpp
+{
+namespace impl
+{
+
+inline const char *
+check_get_default_topic_qos(DDS::ReturnCode_t & status)
```

Comment:
Since the status is a primitive type it should not be passed as a reference.

Same below.